[#5440] Add Ruby 2.7 support

gbp opened the pull request:
## Relevant issue(s)

Depends on #6033 
Fixes #5440 
Linking to https://github.com/mysociety/alaveteli/pull/6118, since there's a conditional in this PR to do this for Ruby 2.7. Older Xapian gem doesn't install on 2.7
Linking to https://github.com/mysociety/sysadmin/issues/1409, since that's an enabler for us running this PR on WDTK

## What does this do?

1. ~~Moves "Rails next" gemfile into subdirectory~~
2. ~~Add a Xapian gemfile for a more recent version with support for Ruby 2.7~~
3. ~~Updates acts as xapian for to support the next Xapain bindings~~
4. Includes Ruby 2.7 in CI

gbp commented:
Had a quick look into this it seems `active_model_otp` isn't compatitible with 2.7 the `rotp` gem has been updated but  `active_model_otp` depends on an older version.

garethrees reviewed (approved):
Would be good to remove the `xapian_upgrade?` conditional stuff since we're going to get all that upgraded imminently.

After that, good to merge.

gbp commented:
Now #6118 has been merged I've rebased. I was able to drop the if `RUBY_VERSION >= '2.7'` conditionals too since we've dropped support for Ruby 2.3 & 2.4
